modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -60,11 +66,97 @@
 
     HashMap<String, HdfsFile> openedFiles;
 
+    private Timer timeReap;
+
+    private Lock lock;
+
+    class TimeReap extends TimerTask {
+        private Map<String, HdfsFile> files;
+        public TimeReap(Map<String, HdfsFile> files) {
+            logger.trace("Create TimeReap");
+            this.files = files;
+        }
+
+        private boolean closeFile(HdfsFile hdfsfile) {
```

Comment:
We don't really. Actually when I started to do some refactor, it would have been possible to have one close. The issue was with archive, because the archiving is started if configured and the magical `bool` flag is flipped. If I wanted to have one `close` for both the timer and for the destination, either I had to deal with archive or just pass an extra parameter as in `closeAllFiles`.

I draw the refactor line at archive, thus the above divided closing.